File: nexus/native/benchmark.cpp
#include <iostream>
#include <vector>
#include <chrono>

bool is_prime(int n) {
    if (n <= 1) return false;
    for (int i = 2; i * i <= n; i++) {
        if (n % i == 0) return false;
    }
    return true;
}

int main() {
    auto start = std::chrono::high_resolution_clock::now();
    
    int count = 0;
    int limit = 500000;
    for (int i = 1; i <= limit; i++) {
        if (is_prime(i)) {
            count++;
        }
    }
    
    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> diff = end - start;
    
    std::cout << "Nexus Native Benchmark (Clang)" << std::endl;
    std::cout << "-------------------------------" << std::endl;
    std::cout << "Primes found up to " << limit << ": " << count << std::endl;
    std::cout << "Time taken: " << diff.count() << " seconds" << std::endl;
    
    return 0;
}

File: nexus/native/benchmark.py
import time

def is_prime(n):
    if n <= 1: return False
    i = 2
    while i * i <= n:
        if n % i == 0: return False
        i += 1
    return True

def main():
    start = time.time()
    count = 0
    limit = 500000
    for i in range(1, limit + 1):
        if is_prime(i):
            count += 1
    end = time.time()
    
    print("Nexus Python Benchmark")
    print("-----------------------")
    print(f"Primes found up to {limit}: {count}")
    print(f"Time taken: {end - start:.6f} seconds")

if __name__ == "__main__":
    main()
